### ISSUE TITLE
Hovering over language keywords crashes server.

### ISSUE THREAD

**Issue by pherrymason** [pherrymason/c3-lsp#9]
**Describe the bug**
Hovering over any language keyword, make the server crash

**To Reproduce**
1. Hover keywords like "if", "import", "module"
2. Server crashes with an infinite loop.

¿Does this happens with any symbol that cannot be found?

**Expected behavior**
Should ignore language keywords.
Make sure it does not breaks when a symbol cannot be found also.

```
2024/05/04 21:58:24.122 DEBUG [C3-LSP.rpc] jsonrpc2: --> request #1: textDocument/hover: {"position":{"character":4,"line":11},"textDocument":{"uri":"file:///path/src/cart.c3"}}
2024/05/04 21:58:24.122 DEBUG [C3-LSP.server] normalized path:/path/src/cart.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.server] HOVER requested on 6726 157
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] | findClosestSymbolDeclaration on doc /path/src/cart.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] | Checking module "cart"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] | findClosestSymbolDeclaration: search in imported module "types"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |_ findSymbolDeclarationInModule: search symbols in module "types" file "/path/src/types.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration on doc /path/src/types.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ Checking module "types"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |_ end searching symbols in module "types" file "/path/src/types.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] | findClosestSymbolDeclaration: search in imported module "system"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |_ findSymbolDeclarationInModule: search symbols in module "system" file "/path/src/system.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration on doc /path/src/system.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ Checking module "system"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration: search in imported module "std::io"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration: search in imported module "std::time"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration: search in imported module "std::time::clock"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration: search in imported module "std::thread"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration: search in imported module "cpu"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ findSymbolDeclarationInModule: search symbols in module "cpu" file "/path/src/cpu.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration on doc /path/src/cpu.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ Checking module "cpu"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration: search in imported module "inst"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |_____ findSymbolDeclarationInModule: search symbols in module "inst" file "/path/src/inst.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |______ findClosestSymbolDeclaration on doc /path/src/inst.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |______ Checking module "inst"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |_____ end searching symbols in module "inst" file "/path/src/inst.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration: search in imported module "debug"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |_____ findSymbolDeclarationInModule: search symbols in module "debug" file "/path/src/debugger/debug.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |______ findClosestSymbolDeclaration on doc /path/src/debugger/debug.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |______ Checking module "debug"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |______ findClosestSymbolDeclaration: search in imported module "raylib"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |______ findClosestSymbolDeclaration: search in imported module "libc"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |_____ end searching symbols in module "debug" file "/path/src/debugger/debug.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ end searching symbols in module "cpu" file "/path/src/cpu.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration: search in imported module "io"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ findSymbolDeclarationInModule: search symbols in module "io" file "/path/src/io.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration on doc /path/src/io.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ Checking module "io"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ end searching symbols in module "io" file "/path/src/io.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration: search in imported module "ppu"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ findSymbolDeclarationInModule: search symbols in module "ppu" file "/path/src/ppu.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration on doc /path/src/ppu.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ Checking module "ppu"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ end searching symbols in module "ppu" file "/path/src/ppu.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration: search in imported module "lcd"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ findSymbolDeclarationInModule: search symbols in module "lcd" file "/path/src/lcd.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration on doc /path/src/lcd.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ Checking module "lcd"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration: search in imported module "raylib"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ end searching symbols in module "lcd" file "/path/src/lcd.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration: search in imported module "input"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ findSymbolDeclarationInModule: search symbols in module "input" file "/path/src/input.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration on doc /path/src/input.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ Checking module "input"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration: search in imported module "raylib"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ end searching symbols in module "input" file "/path/src/input.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |__ findClosestSymbolDeclaration: search in imported module "cart"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |___ findSymbolDeclarationInModule: search symbols in module "cart" file "/path/src/cart.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration on doc /path/src/cart.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ Checking module "cart"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |____ findClosestSymbolDeclaration: search in imported module "mappers"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |_____ findSymbolDeclarationInModule: search symbols in module "mappers" file "/path/src/mappers/mapper.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |______ findClosestSymbolDeclaration on doc /path/src/mappers/mapper.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |______ Checking module "mappers"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ findClosestSymbolDeclaration on doc /path/src/mappers/mcb5.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ Checking module "mappers::mcb5"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ findClosestSymbolDeclaration: search in imported module "types"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ findSymbolDeclarationInModule: search symbols in module "types" file "/path/src/types.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ findClosestSymbolDeclaration on doc /path/src/types.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ Checking module "types"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ end searching symbols in module "types" file "/path/src/types.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ findClosestSymbolDeclaration: search in imported module "cart"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ findSymbolDeclarationInModule: search symbols in module "cart" file "/path/src/cart.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ findClosestSymbolDeclaration on doc /path/src/cart.c3
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ Checking module "cart"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ findClosestSymbolDeclaration: search in imported module "system"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ findSymbolDeclarationInModule: search symbols in module "system" file "/path/src/system.c3"
2024/05/04 21:58:24.122 DEBUG [C3-LSP.parser] |________ findClosestSymbolDeclaration on doc /path/src/system.c3
````

**Comment by pherrymason** [comment]
Fixed infinite loop in commit #[cea2ae6](https://github.com/pherrymason/c3-lsp/commit/cea2ae6406a07a55895bdcac84e9ed8999beefab)

**Comment by pherrymason** [comment]
Ignore keywords fixed in [0a390fd](https://github.com/pherrymason/c3-lsp/commit/0a390fd915fc3c46fb12cac4a48855331278d088)

**Comment by pherrymason** [comment]
Added missing tests on [c53ff06](https://github.com/pherrymason/c3-lsp/commit/c53ff06a7dce2af78a95f9c2d0510b60dc36329b)
